Makefile

```diff
@@ -548,103 +569,125 @@ $(TOOLCHAIN):
 	git config --global core.autocrlf false
 	mkdir -p $(TOOLCHAIN)
 
+$(PATCH_DIR):
+ifeq ($(GCC_VERSION),4.9.2)
+	cd $(XTDLP)/$(GCC_DIR)
+	patch -p1 < $(PATCH_DIR)/0001-WIP-don-t-bring-extra-u-int_least32_t-into-std.patch
+endif
+
 # GMP
 $(XTDLP)/$(GMP_DIR): $(XTDLP)/$(GMP_TAR)
 	mkdir -p $(XTDLP)/$(GMP_DIR)
 	$(UNTAR) $(XTDLP)/$(GMP_TAR) -C $(XTDLP)/$(GMP_DIR)
-	mv $(XTDLP)/$(GMP_DIR)/gmp-*/* $(XTDLP)/$(GMP_DIR)
+	rm -rf $(XTDLP)/$(GMP_DIR)/build
 
 $(XTDLP)/$(GMP_DIR)/build: $(XTDLP)/$(GMP_DIR)
 	mkdir -p $(XTDLP)/$(GMP_DIR)/build/
-	cd $(XTDLP)/$(GMP_DIR)/build/; ../configure --prefix=$(XTBP)/gmp --disable-shared --enable-static --build=$(BUILD_TARGET) --host=$(BUILD_TARGET)
-	make -C $(XTDLP)/$(GMP_DIR)/build/	
+	cd $(XTDLP)/$(GMP_DIR)/build/; ../$(CONFIGURE) --prefix=$(XTBP)/gmp --disable-shared --enable-static --build=$(BUILD_TARGET) --host=$(BUILD_TARGET)
+	$(MAKE) -C $(XTDLP)/$(GMP_DIR)/build/
+	@touch $(XTDLP)/$(GMP_DIR)/build
 
 $(XTBP)/gmp: $(XTDLP)/$(GMP_DIR)/build
-	make install -C $(XTDLP)/$(GMP_DIR)/build/
+	$(MAKE) $(INSTALL) -C $(XTDLP)/$(GMP_DIR)/build/
+	@touch $(XTBP)/gmp
 
-# MPFR	
+# MPFR
 $(XTDLP)/$(MPFR_DIR): $(XTDLP)/$(MPFR_TAR)
 	mkdir -p $(XTDLP)/$(MPFR_DIR)
 	$(UNTAR) $(XTDLP)/$(MPFR_TAR) -C $(XTDLP)/$(MPFR_DIR)
-	mv $(XTDLP)/$(MPFR_DIR)/mpfr-*/* $(XTDLP)/$(MPFR_DIR)
+	rm -rf $(XTDLP)/$(MPFR_DIR)/build
 
 $(XTDLP)/$(MPFR_DIR)/build: $(XTDLP)/$(MPFR_DIR)
 	mkdir -p $(XTDLP)/$(MPFR_DIR)/build
-	cd $(XTDLP)/$(MPFR_DIR)/build/; ../configure --prefix=$(XTBP)/mpfr --with-gmp=$(XTBP)/gmp --disable-shared --enable-static --build=$(BUILD_TARGET) --host=$(BUILD_TARGET)
-	make -C $(XTDLP)/$(MPFR_DIR)/build/	
+	cd $(XTDLP)/$(MPFR_DIR)/build/; ../$(CONFIGURE) --prefix=$(XTBP)/mpfr --with-gmp=$(XTBP)/gmp --disable-shared --enable-static --build=$(BUILD_TARGET) --host=$(BUILD_TARGET)
+	$(MAKE) -C $(XTDLP)/$(MPFR_DIR)/build/
+	@touch $(XTDLP)/$(MPFR_DIR)/build
```

Comment:
Can you please explain why this is needed? 
wouldn't `mkdir -p $(XTDLP)/$(MPFR_DIR)/build` be enough?